From 8ed96726603a59969b99e4ea30dbd9b06955084b Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Sun, 09 Feb 2014 13:38:03 +0000
Subject: libbb: don't die if crypt() returns NULL

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
diff --git a/libbb/pw_encrypt.c b/libbb/pw_encrypt.c
index 39ffa08..bfc7030 100644
--- a/libbb/pw_encrypt.c
+++ b/libbb/pw_encrypt.c
@@ -142,7 +142,14 @@ char* FAST_FUNC pw_encrypt(const char *clear, const char *salt, int cleanup)
 
 char* FAST_FUNC pw_encrypt(const char *clear, const char *salt, int cleanup)
 {
-	return xstrdup(crypt(clear, salt));
+	char *s;
+
+	s = crypt(clear, salt);
+	/*
+	 * glibc used to return "" on malformed salts (for example, ""),
+	 * but since 2.17 it returns NULL.
+	 */
+	return xstrdup(s ? s : "");
 }
 
 #endif
--
cgit v0.9.1
