From 6c6d37ee4526f77fe07e46f05b0378ce43421e84 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Tue, 29 Jul 2014 15:00:30 +0000
Subject: top: fix memset length (sizeof(ptr) vs sizeof(array) problem)

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
diff --git a/procps/top.c b/procps/top.c
index 62f9421..3d67c3c 100644
--- a/procps/top.c
+++ b/procps/top.c
@@ -536,7 +536,7 @@ static void parse_meminfo(unsigned long meminfo[MI_MAX])
 	FILE *f;
 	int i;
 
-	memset(meminfo, 0, sizeof(meminfo));
+	memset(meminfo, 0, sizeof(meminfo[0]) * MI_MAX);
 	f = xfopen_for_read("meminfo");
 	while (fgets(buf, sizeof(buf), f) != NULL) {
 		char *c = strchr(buf, ':');
@@ -550,7 +550,6 @@ static void parse_meminfo(unsigned long meminfo[MI_MAX])
 	fclose(f);
 }
 
-
 static unsigned long display_header(int scr_width, int *lines_rem_p)
 {
 	char scrbuf[100]; /* [80] was a bit too low on 8Gb ram box */
--
cgit v0.9.1
