From 307d26c0ebfb0d22a81070379675f6287f6e9d95 Mon Sep 17 00:00:00 2001
From: Aaro Koskinen <aaro.koskinen@iki.fi>
Date: Tue, 23 Sep 2014 20:58:18 +0000
Subject: less: make -E work

function                                             old     new   delta
buffer_print                                          71     104     +33

Signed-off-by: Aaro Koskinen <aaro.koskinen@iki.fi>
Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
diff --git a/miscutils/less.c b/miscutils/less.c
index c5820d3..554e546 100644
--- a/miscutils/less.c
+++ b/miscutils/less.c
@@ -802,11 +802,18 @@ static void buffer_print(void)
 	unsigned i;
 
 	move_cursor(0, 0);
-	for (i = 0; i <= max_displayed_line; i++)
+	for (i = 0; i <= max_displayed_line; i++) {
 		if (pattern_valid)
 			print_found(buffer[i]);
 		else
 			print_ascii(buffer[i]);
+	}
+	if ((option_mask32 & FLAG_E)
+	 && eof_error <= 0
+	 && (max_fline - cur_fline) <= max_displayed_line
+	) {
+		less_exit(EXIT_SUCCESS);
+	}
 	status_print();
 }
 
--
cgit v0.9.1
